# HG changeset patch
# Parent 532318f98505734384e742ee975e00a33d901ee6
misc/156659: periodic/daily/800.scrub-zfs fails on pool names containing a hyphen

diff -r 532318f98505 etc/periodic/daily/800.scrub-zfs
--- a/etc/periodic/daily/800.scrub-zfs
+++ b/etc/periodic/daily/800.scrub-zfs
@@ -27,10 +27,13 @@
 	fi
 
 	for pool in ${daily_scrub_zfs_pools}; do
+
+		_pool_name=$(echo $pool |cut -f1 -d: |sed 's/://')
+
 		# sanity check
-		_status=$(zpool list "${pool}" 2> /dev/null)
+		_status=$(zpool list "${_pool_name}" 2> /dev/null)
 		if [ $? -ne 0 ]; then
-			echo "   WARNING: pool '${pool}' specified in"
+			echo "   WARNING: pool ${_pool_name} specified in"
 			echo "            '/etc/periodic.conf:daily_scrub_zfs_pools'"
 			echo "            does not exist"
 			continue
@@ -38,7 +41,7 @@
 		_status=${_status##*$newline}
 		case ${_status} in
 		*FAULTED*)
-			echo "Skipping faulted pool: ${pool}"
+			echo "Skipping faulted pool: ${_pool_name}"
 			continue ;;
 		esac
 
@@ -46,17 +49,17 @@
 		rc=0
 
 		# determine how many days shall be between scrubs
-		eval _pool_threshold=\${daily_scrub_zfs_${pool}_threshold}
-		if [ -z "${_pool_threshold}" ];then
+		_pool_threshold=$(echo $pool |cut -f2 -d: |grep -v [aA-zZ])
+		if [ ! -n "${_pool_threshold}" ]; then
 			_pool_threshold=${daily_scrub_zfs_default_threshold}
 		fi
 
-		_last_scrub=$(zpool history ${pool} | \
-		    egrep "^[0-9\.\:\-]{19} zpool scrub ${pool}\$" | tail -1 |\
+		_last_scrub=$(zpool history ${_pool_name} | \
+		    egrep "^[0-9\.\:\-]{19} zpool scrub ${_pool_name}\$" | tail -1 |\
 		    cut -d ' ' -f 1)
 		if [ -z "${_last_scrub}" ]; then
 			# creation time of the pool if no scrub was done
-			_last_scrub=$(zpool history ${pool} | \
+			_last_scrub=$(zpool history ${_pool_name} | \
 			    sed -ne '2s/ .*$//p')
 		fi
 
@@ -64,27 +67,27 @@
 		_scrub_diff=$(expr -e \( $(date +%s) - \
 		    $(date -j -f %F.%T ${_last_scrub} +%s) \) / 60 / 60 / 24)
 		if [ ${_scrub_diff} -lt ${_pool_threshold} ]; then
-			echo "   skipping scrubbing of pool '${pool}':"
+			echo "   skipping scrubbing of pool '${_pool_name}':"
 			echo "      last scrubbing is ${_scrub_diff} days ago, threshold is set to ${_pool_threshold} days"
 			continue
 		fi
 
-		_status="$(zpool status ${pool} | grep scrub:)"
+		_status="$(zpool status ${_pool_name} | grep scrub:)"
 		case "${_status}" in
 			*"scrub in progress"*)
-				echo "   scrubbing of pool '${pool}' already in progress, skipping:"
+				echo "   scrubbing of pool '${_pool_name}' already in progress, skipping:"
 				;;
 			*"none requested"*)
-				echo "   starting first scrub (since reboot) of pool '${pool}':"
-				zpool scrub ${pool}
+				echo "   starting first scrub (since reboot) of pool '${_pool_name}':"
+				zpool scrub ${_pool_name}
 				;;
 			*)
-				echo "   starting scrub of pool '${pool}':"
-				zpool scrub ${pool}
+				echo "   starting scrub of pool '${_pool_name}':"
+				zpool scrub ${_pool_name}
 				;;
 		esac
 
-		echo "      consult 'zpool status ${pool}' for the result"
+		echo "      consult 'zpool status ${_pool_name}' for the result"
 	done
 	;;
 
